src/EventListener/CatchAttributesListener.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace FOS\HttpCacheBundle\EventListener;
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\ConfigurationInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\ControllerResolver;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\Kernel;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+if (Kernel::MAJOR_VERSION >= 5) {
+    class_alias(RequestEvent::class, 'FOS\HttpCacheBundle\EventListener\AttributeRequestEvent');
+} else {
+    class_alias(GetResponseEvent::class, 'FOS\HttpCacheBundle\EventListener\AttributeRequestEvent');
+}
+
```

Comment:
can you please add a phpdoc here that explains that this is about handling the php 8 attributes?

it has been a while that i looked into the codebase - how are the doctrine annotations provided into the request attributes? is that symfony itself doing it? does it not do that for php 8 annotations?

Reply:
Yes, I'll add phpdoc for this class :) 

In Symfony, If we look the Routing component, we have a class named AnnotationClassLoader who load doctrine annotations or php8 attributes. They inject in the constructor Doctrine\Common\Annotations\Reader to fetch doctrine annotations.... And they use a check on PHP_VERSION_ID and not on method_exists to fetch php8 attributes :-D